Digger CLI - invalid memory address or nil pointer dereference

I have issue in the workflow of digger apply (I have another issues described in https://github.com/diggerhq/digger/issues/1110 , but this one is when I don't use upload-plan-destination: github)
Everything runs correctly, apply succeeds, Github Application creates the comment, but the pipeline crashes in the end because of error in digger.go:

```
Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

no $DIGGER_OUT file at: /__w/_temp/digger-out.log
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1465278]

goroutine 1 [running]:
github.com/diggerhq/digger/cli/pkg/digger.RunJobs({0xc001197ae0, 0x1, 0xc000c1c630?}, {0x24d1f80, 0xc000be6db0}, {0x24b2be0, 0xc000be6db0}, {0x24c51f0, 0xc000126440}, {0x24be0a8, ...}, ...)
	/home/runner/work/digger/digger/cli/pkg/digger/digger.go:137 +0x3b8
main.gitHubCI({0x24c51f0, 0xc000126440}, {0x24c4f20, 0xc000acf9b0}, {0x24c4f50, 0xc000bc6360}, {0x24b2b20?, 0xc000ae25d0})
	/home/runner/work/digger/digger/cli/cmd/digger/main.go:158 +0x1247
main.main()
	/home/runner/work/digger/digger/cli/cmd/digger/main.go:852 +0xa5c
Error: Error: failed to run script step: command terminated with non-zero exit code: error executing command [sh -e /__w/_temp/614250b0-c104-11ee-93dc-e7b51bfeb18a.sh], exit code 2
Error: Process completed with exit code 1.
Error: Executing the custom container implementation failed. Please contact your self hosted runner administrator.
```

To compare, successfull log from plan:
```
Plan: 1 to add, 0 to change, 0 to destroy.
─────────────────────────────────────────────────────────────────────────────
Saved the plan to:
/__w/XXX/XXX/YYY.tfplan

To perform exactly these actions, run the following command to apply:
 ...
Running command: terraform [show -no-color -json ...]
no $DIGGER_OUT file at: /__w/_temp/digger-out.log
No plan policies found, succeeding
Digger finished successfully
Run if [[ "workflow_dispatch" == "issue_comment" ]]; then
Run '/home/runner/k8s/index.js'
```

Using self hosted backend v3.19.0, action of the same version. I don’t know how to replicate the issue. Ideally, the cli would have more info and debug logs to identify the cause, I can try to add those and build my own version.

Also seeing a similar issue; no plan/applies work with action + backend versions 0.3.17+ using the GithubCI. 

```
18:34:15 locking.go:248: Using NoOp lock provider.
18:34:15 root.go:113: Lock provider has been created successfully
18:34:15 main.go:39: Using GitHub.
18:34:15 main.go:76: GitHub context parsed successfully
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x70 pc=0x193e813]

goroutine 1 [running]:
main.gitHubCI({0x2597658, 0x364bf00}, {0x2596c98, 0xc0009bfbb0}, {0x2596cc8, 0xc000b464e0}, {0x25842c0?, 0xc000625de8})
	/home/runner/work/_actions/diggerhq/digger/latest/cli/cmd/digger/main.go:128 +0xc93
main.glob..func2(0xc000b4e200?, {0x1f565b4?, 0x4?, 0x1f565b8?})
	/home/runner/work/_actions/diggerhq/digger/latest/cli/cmd/digger/default.go:23 +0x205
github.com/spf13/cobra.(*Command).execute(0x35e3c40, {0x364bf00, 0x0, 0x0})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:987 +0xaa3
github.com/spf13/cobra.(*Command).ExecuteC(0x35e44e0)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
main.main()
	/home/runner/work/_actions/diggerhq/digger/latest/cli/cmd/digger/main.go:815 +0xf4
Error: Process completed with exit code 2.
```

--- a/cli/pkg/digger/digger.go
+++ b/cli/pkg/digger/digger.go
@@ -173,6 +173,7 @@ func UpdateStatusComment(jobs []scheduler.SerializedJob, prNumber int, prService
 
 		fmt.Printf("job details: %v | %v\n", jobSpec.Commands, isPlan)
 		spew.Dump(jobSpec.Commands)
+		fmt.Printf("%v ---- %v", jobSpec.Commands[0], jobSpec.Commands[0] == "digger plan")
 
 		message = message + fmt.Sprintf("<!-- PROJECTHOLDER %v -->\n", job.ProjectName)
 		message = message + fmt.Sprintf("%v **%v** <a href='%v'>%v</a>%v\n", job.Status.ToEmoji(), jobSpec.ProjectName, *job.WorkflowRunUrl, job.Status.ToString(), job.ResourcesSummaryString(isPlan))

--- a/cli/pkg/digger/digger.go
+++ b/cli/pkg/digger/digger.go
@@ -135,7 +135,12 @@ func RunJobs(
 		currentJob := jobs[0]
 		repoNameForBackendReporting := strings.ReplaceAll(currentJob.Namespace, "/", "-")
 		projectNameForBackendReporting := currentJob.ProjectName
-		planSummary := exectorResults[0].PlanResult.PlanSummary
+		// TODO: handle the apply result summary as well to report it to backend. Possibly reporting changed resources as well
+		// Some kind of generic terraform operation summary might need to be introduced
+		planSummary := terraform.PlanSummary{}
+		if exectorResults[0].PlanResult != nil {
+			planSummary = exectorResults[0].PlanResult.PlanSummary
+		}
 		prNumber := *currentJob.PullRequestNumber
 		batchResult, err := backendApi.ReportProjectJobStatus(repoNameForBackendReporting, projectNameForBackendReporting, batchId, "succeeded", time.Now(), &planSummary)
 		if err != nil {

--- a/cli/pkg/digger/digger.go
+++ b/cli/pkg/digger/digger.go
@@ -170,6 +170,8 @@ func UpdateStatusComment(jobs []scheduler.SerializedJob, prNumber int, prService
 			log.Printf("Failed to convert unmarshall Serialized job")
 		}
 
+		fmt.Printf("job details: %v | %v", jobSpec.Commands, isPlan)
+
 		message = message + fmt.Sprintf("<!-- PROJECTHOLDER %v -->\n", job.ProjectName)
 		message = message + fmt.Sprintf("%v **%v** <a href='%v'>%v</a>%v\n", job.Status.ToEmoji(), jobSpec.ProjectName, *job.WorkflowRunUrl, job.Status.ToString(), job.ResourcesSummaryString(isPlan))
 		message = message + fmt.Sprintf("<!-- PROJECTHOLDEREND %v -->\n", job.ProjectName)

--- a/cli/pkg/digger/digger.go
+++ b/cli/pkg/digger/digger.go
@@ -165,11 +165,11 @@ func UpdateStatusComment(jobs []scheduler.SerializedJob, prNumber int, prService
 	message := ":construction_worker: Jobs status:\n\n"
 	for _, job := range jobs {
 		var jobSpec orchestrator.JobJson
-		isPlan := jobSpec.IsPlan()
 		err := json.Unmarshal(job.JobString, &jobSpec)
 		if err != nil {
 			log.Printf("Failed to convert unmarshall Serialized job")
 		}
+		isPlan := jobSpec.IsPlan()
 
 		fmt.Printf("job details: %v | %v\n", jobSpec.Commands, isPlan)
 		spew.Dump(jobSpec.Commands)

--- a/cli/pkg/digger/digger.go
+++ b/cli/pkg/digger/digger.go
@@ -3,7 +3,6 @@ package digger
 import (
 	"errors"
 	"fmt"
-	"github.com/davecgh/go-spew/spew"
 	"log"
 	"os"
 	"path"
@@ -170,10 +169,6 @@ func UpdateStatusComment(jobs []scheduler.SerializedJob, prNumber int, prService
 			log.Printf("Failed to convert unmarshall Serialized job")
 		}
 		isPlan := jobSpec.IsPlan()
-
-		fmt.Printf("job details: %v | %v\n", jobSpec.Commands, isPlan)
-		spew.Dump(jobSpec.Commands)
-		fmt.Printf("%v ---- %v", jobSpec.Commands[0], jobSpec.Commands[0] == "digger plan")
 
 		message = message + fmt.Sprintf("<!-- PROJECTHOLDER %v -->\n", job.ProjectName)
 		message = message + fmt.Sprintf("%v **%v** <a href='%v'>%v</a>%v\n", job.Status.ToEmoji(), jobSpec.ProjectName, *job.WorkflowRunUrl, job.Status.ToString(), job.ResourcesSummaryString(isPlan))

--- a/cli/pkg/digger/digger.go
+++ b/cli/pkg/digger/digger.go
@@ -3,6 +3,7 @@ package digger
 import (
 	"errors"
 	"fmt"
+	"github.com/davecgh/go-spew/spew"
 	"log"
 	"os"
 	"path"
@@ -170,7 +171,8 @@ func UpdateStatusComment(jobs []scheduler.SerializedJob, prNumber int, prService
 			log.Printf("Failed to convert unmarshall Serialized job")
 		}
 
-		fmt.Printf("job details: %v | %v", jobSpec.Commands, isPlan)
+		fmt.Printf("job details: %v | %v\n", jobSpec.Commands, isPlan)
+		spew.Dump(jobSpec.Commands)
 
 		message = message + fmt.Sprintf("<!-- PROJECTHOLDER %v -->\n", job.ProjectName)
 		message = message + fmt.Sprintf("%v **%v** <a href='%v'>%v</a>%v\n", job.Status.ToEmoji(), jobSpec.ProjectName, *job.WorkflowRunUrl, job.Status.ToString(), job.ResourcesSummaryString(isPlan))

--- a/cli/pkg/digger/digger.go
+++ b/cli/pkg/digger/digger.go
@@ -163,15 +163,15 @@ func UpdateStatusComment(jobs []scheduler.SerializedJob, prNumber int, prService
 
 	message := ":construction_worker: Jobs status:\n\n"
 	for _, job := range jobs {
-
 		var jobSpec orchestrator.JobJson
+		isPlan := jobSpec.IsPlan()
 		err := json.Unmarshal(job.JobString, &jobSpec)
 		if err != nil {
 			log.Printf("Failed to convert unmarshall Serialized job")
 		}
 
 		message = message + fmt.Sprintf("<!-- PROJECTHOLDER %v -->\n", job.ProjectName)
-		message = message + fmt.Sprintf("%v **%v** <a href='%v'>%v</a>%v\n", job.Status.ToEmoji(), jobSpec.ProjectName, *job.WorkflowRunUrl, job.Status.ToString(), job.ResourcesSummaryString())
+		message = message + fmt.Sprintf("%v **%v** <a href='%v'>%v</a>%v\n", job.Status.ToEmoji(), jobSpec.ProjectName, *job.WorkflowRunUrl, job.Status.ToString(), job.ResourcesSummaryString(isPlan))
 		message = message + fmt.Sprintf("<!-- PROJECTHOLDEREND %v -->\n", job.ProjectName)
 	}
 

--- a/libs/orchestrator/aws.go
+++ b/libs/orchestrator/aws.go
@@ -42,7 +42,7 @@ func populateKeys(envs map[string]string, provider stscreds.WebIdentityRoleProvi
 func (job *Job) PopulateAwsCredentialsEnvVarsForJob() error {
 
 	if job.StateEnvProvider != nil {
-		log.Printf("Project-level AWS role detected, Assuming role: %v for project run: %v", job.ProjectName)
+		log.Printf("Project-level AWS role detected, Assuming role for project: %v", job.ProjectName)
 		var err error
 		backendConfigArgs, err := populateretrieveBackendConfigArgs(*job.StateEnvProvider)
 		if err != nil {

--- a/libs/orchestrator/json_models.go
+++ b/libs/orchestrator/json_models.go
@@ -1,5 +1,7 @@
 package orchestrator
 
+import "slices"
+
 type StepJson struct {
 	Action    string   `json:"action"`
 	ExtraArgs []string `json:"extraArgs"`
@@ -23,6 +25,14 @@ type JobJson struct {
 	Namespace         string            `json:"namespace"`
 	StateEnvVars      map[string]string `json:"stateEnvVars"`
 	CommandEnvVars    map[string]string `json:"commandEnvVars"`
+}
+
+func (j *JobJson) IsPlan() bool {
+	return slices.Contains(j.Commands, "digger plan")
+}
+
+func (j *JobJson) IsApply() bool {
+	return slices.Contains(j.Commands, "digger apply")
 }
 
 func JobToJson(job Job) JobJson {

--- a/libs/orchestrator/json_models_test.go
+++ b/libs/orchestrator/json_models_test.go
@@ -1,0 +1,32 @@
+package orchestrator
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestIsPlanForDiggerPlanJobCorrect(t *testing.T) {
+	j := JobJson{
+		ProjectName:      "project.Name",
+		ProjectDir:       "project.Dir",
+		ProjectWorkspace: "workspace",
+		Terragrunt:       false,
+		Commands:         []string{"run echo 'hello", "digger plan"},
+		EventName:        "issue_comment",
+	}
+	assert.True(t, j.IsPlan())
+	assert.False(t, j.IsApply())
+}
+
+func TestIsApplyForDiggerApplyJobCorrect(t *testing.T) {
+	j := JobJson{
+		ProjectName:      "project.Name",
+		ProjectDir:       "project.Dir",
+		ProjectWorkspace: "workspace",
+		Terragrunt:       false,
+		Commands:         []string{"digger apply"},
+		EventName:        "issue_comment",
+	}
+	assert.True(t, j.IsApply())
+	assert.False(t, j.IsPlan())
+}

--- a/libs/orchestrator/scheduler/models.go
+++ b/libs/orchestrator/scheduler/models.go
@@ -86,8 +86,8 @@ type SerializedBatch struct {
 	Jobs         []SerializedJob   `json:"jobs"`
 }
 
-func (s *SerializedJob) ResourcesSummaryString() string {
-	if s.Status == DiggerJobSucceeded {
+func (s *SerializedJob) ResourcesSummaryString(isPlan bool) string {
+	if isPlan && s.Status == DiggerJobSucceeded {
 		return fmt.Sprintf(" [Resources: %v to create, %v to update, %v to delete]", s.ResourcesCreated, s.ResourcesUpdated, s.ResourcesDeleted)
 	} else {
 		return "..."

--- a/libs/orchestrator/scheduler/models.go
+++ b/libs/orchestrator/scheduler/models.go
@@ -87,7 +87,11 @@ type SerializedBatch struct {
 }
 
 func (s *SerializedJob) ResourcesSummaryString(isPlan bool) string {
-	if isPlan && s.Status == DiggerJobSucceeded {
+	if !isPlan {
+		return ""
+	}
+
+	if s.Status == DiggerJobSucceeded {
 		return fmt.Sprintf(" [Resources: %v to create, %v to update, %v to delete]", s.ResourcesCreated, s.ResourcesUpdated, s.ResourcesDeleted)
 	} else {
 		return "..."

